workflow/controller/controller.go

```diff
@@ -1206,3 +1215,24 @@ func (wfc *WorkflowController) newWorkflowTaskSetInformer() wfextvv1alpha1.Workf
 		})
 	return informer
 }
+
+func (wfc *WorkflowController) newArtGCTaskInformer() wfextvv1alpha1.WorkflowArtifactGCTaskInformer {
+	informer := externalversions.NewSharedInformerFactoryWithOptions(
+		wfc.wfclientset,
+		workflowTaskSetResyncPeriod,
+		externalversions.WithNamespace(wfc.GetManagedNamespace()),
+		externalversions.WithTweakListOptions(func(x *metav1.ListOptions) {
+			r := util.InstanceIDRequirement(wfc.Config.InstanceID)
+			x.LabelSelector = r.String()
+		})).Argoproj().V1alpha1().WorkflowArtifactGCTasks()
+	informer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(old, new interface{}) {
+				key, err := cache.MetaNamespaceKeyFunc(new)
+				if err == nil {
+					wfc.wfQueue.AddRateLimited(key)
```

Comment:
Why does it need to be re-queued? Can it directly  call `garbageCollectArtifacts`?

Reply:
Maybe I could try that. I was imitating the newWorkflowTaskSetInformer() above. But one thing I'm not sure of is - if I need `execWf` don't I need to call all of the other logic in operate()? Also, I think there's some artifact repository resolution logic I'm dependent on from operate(). So, I tend to think it would be better to go through the processing logic.

Reply:
we will keep it here for MVP and refactor  a separate module to preprocessing the workflow

Reply:
Thanks. So, so are you think we'd have maybe a new method for pre-processing which would get called both from operate() and from here? I like that idea.